File: pangath_properties/events/issue.py
import frappe

@frappe.whitelist()
def create_maintenance_schedule(doc):
    issue_doc=frappe.get_doc("Issue",doc)
    main_doc=frappe.new_doc("Maintenance Schedule")
    main_doc.customer=issue_doc.customer
    main_doc.custom_tenancy_contract=issue_doc.custom_tenancy_contract
    main_doc.custom_property=issue_doc.property
    main_doc.custom_unit_no=issue_doc.unit
    main_doc.issue=issue_doc.name
    if issue_doc.custom_tenancy_contract:
        main_doc.company=frappe.db.get_value("Tenancy Contract",issue_doc.custom_tenancy_contract,"company")
    if main_doc:
        return main_doc



def validate(doc,method):
    doc.subject = str(doc.unit) + "-" + str(doc.issue_type) + "-" + str(doc.custom_request_date)

File: pangath_properties/public/js/issue.js
frappe.ui.form.on("Issue", {
	onload: function (frm) {
		frm.set_query('unit', function (doc) {
			let filters = []
			if (frm.doc.property) {
				filters.push(["Unit", "property", "in", frm.doc.property])
			}
			return {
				"filters": filters
			};
		})
	},
	refresh: function(frm) {
		// hide + icon of  Task from connection 
		setTimeout(() => {
			$("[data-doctype='Task']").find("button").hide();
		  }, 10);
		  frm.add_custom_button(__("Create Maintenance Schedule"), function(){
			frappe.call({
				method: 'real_estate.events.issue.create_maintenance_schedule',
				args: { 'doc': frm.doc.name },
				callback: function(r) {
					if (r && r.message) {
						var doc = frappe.model.sync(r.message);
					frappe.set_route("Form", r.message.doctype, r.message.name);
					}
				}
			});
		  },);
	},
	unit:function(frm){
		frappe.db.get_value("Tenancy Contract",filters = {"property_name":frm.doc.property,"unit_number":frm.doc.unit},["name","name_of_tenant"]).then(r =>{
			frm.set_value("custom_tenancy_contract",r.message.name)
			frm.set_value("customer",r.message.name_of_tenant)

		})
	},
	custom_tenancy_contract:function(frm){
		frappe.db.get_value("Tenancy Contract",frm.doc.custom_tenancy_contract,["property_name","unit_number","name_of_tenant"]).then(r =>{
			frm.set_value("property",r.message.property_name)
			frm.set_value("customer",r.message.name_of_tenant)
			frm.set_value("unit",r.message.unit_number)
		})
	}
});